test/metabase/query_processor/streaming/csv_test.clj

```diff
@@ -55,3 +57,23 @@
             ["4" "Simcha Yan"          "2014-01-01T08:30:00"]
             ["5" "Quentin Sören"       "2014-10-03T17:30:00"]]
            (parse-and-sort-csv result)))))
+
+(defn- csv-export
+  "Given a seq of result rows, write it as a CSV, then read the CSV and return the resulting data."
+  [rows]
+  (with-open [bos (ByteArrayOutputStream.)
+              os  (BufferedOutputStream. bos)]
+    (let [results-writer (qp.si/streaming-results-writer :csv os)]
+      (qp.si/begin! results-writer {:data {:ordered-cols []}} {})
+      (doall (map-indexed
+              (fn [i row] (qp.si/write-row! results-writer row i [] {}))
+              rows))
+      (qp.si/finish! results-writer {:row_count (count rows)}))
+    (let [bytea (.toByteArray bos)]
+      (rest (csv/read-csv (String. bytea))))))
+
+(deftest lazy-seq-realized-test
+  (testing "Lazy seqs within rows are automatically realized during exports (#26261)"
+    (let [row (first (csv-export [[(lazy-seq [1 2 3])]]))]
```

Comment:
Is testing the case when a `ZonedDateTime` is an element of the lazy list interesting?